.github/workflows/scripts/slack_gem_notifications/notifications_methods.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require 'time'
+require "httparty"
+
+def check_for_updates(watched_gems)
+  return if gem_list_empty?(watched_gems)
+  watched_gems.each do |gem_name|
+    return unless is_valid_gem?(gem_name)
```

Comment:
Verifying that a good HTTP response was received before attempting to look at the version info is a great addition! Nice job. There is a drawback to the current validation approach, though. The exact same GET request is being made twice; once for validation and once to go through the version data. Can you think of a way to still perform the important validation while only performing a single GET request?